documentation/blog/2024-01-01-whats-new-in-svelte-january-2024.md

```diff
@@ -0,0 +1,104 @@
+---
+title: "What's new in Svelte: January 2024"
+description: 'SvelteKit 2 and a much-improved $state rune'
+author: Dani Sandoval
+authorURL: https://dreamindani.com
+---
+
+Happy New Year! It's been a busy month for the Svelte maintainers - with tons of new features dropping in the Svelte 5 preview and the [release of SvelteKit 2](https://svelte.dev/blog/sveltekit-2)!
+
+You can find all the new features in both projects below, along with a bunch of resources and sites built with Svelte in the Community Showcase.
+
+Let's jump in...
+
+## What's new in SvelteKit (2.0 and more!)
+
+- `resolvePath` has been replaced by `resolveRoute` in `$app/paths`. Use it to populate a route ID with params to resolve a pathname (**1.29.0**, [Docs](https://kit.svelte.dev/docs/modules#app-paths-resolveroute), [#11261](https://github.com/sveltejs/kit/pull/11261))
+- `response.arrayBuffer()` will now be inlined during SSR (**1.30.0**, [Docs](https://kit.svelte.dev/docs/load#making-fetch-requests), [#10535](https://github.com/sveltejs/kit/pull/10535))
+- [SvelteKit 2.0.0](https://github.com/sveltejs/kit/blob/main/packages/kit/CHANGELOG.md#200) adds:
+  - `untrack` to `load` to opt-out of invalidation ([Docs](https://kit.svelte.dev/docs/load#rerunning-load-functions-untracking-dependencies), [#11311](https://github.com/sveltejs/kit/pull/11311))
+  - shallow routing to create history entries without navigating ([Docs](https://kit.svelte.dev/docs/shallow-routing), [#11307](https://github.com/sveltejs/kit/pull/11307))
+  - html typings ([#11222](https://github.com/sveltejs/kit/pull/11222))
+  - redacted internal stack traces when reporting config errors ([#11292](https://github.com/sveltejs/kit/pull/11292))
+  - fine grained invalidation of search params ([Docs](https://kit.svelte.dev/docs/load#rerunning-load-functions), [#11258](https://github.com/sveltejs/kit/pull/11258))
+
+You can find a migration guide for SvelteKit 2.0 [on the SvelteKit docs](https://kit.svelte.dev/docs/migrating-to-sveltekit-2). Things should be pretty seamless with the `svelte-migrate` command doing much (if not all) for you!
+
+
+## What's new in Svelte
+
+With [Svelte 5 in preview](https://svelte-5-preview.vercel.app/docs/introduction), Svelte 4 (`@latest`) has only been getting bug fixes - with its current version at `4.2.8`. The updates below are from version 5's preview branch:
+
```

Comment:
`$state.frozen` was just added: https://github.com/sveltejs/svelte/pull/9851

will be a part of `5.0.0-next.27`